File: sdk/python/siput_sdk/wallet.py
import secrets
import hashlib


class Wallet:
    @staticmethod
    def create() -> dict:
        entropy = secrets.token_bytes(32)
        address = hashlib.sha256(entropy).hexdigest()[:40]
        return {
            'private_key': entropy.hex(),
            'address': address,
        }

    @staticmethod
    def sign_transaction(private_key: str, tx_data: dict) -> dict:
        payload = f"{tx_data!r}:{private_key}".encode('utf-8')
        signature = hashlib.sha256(payload).hexdigest()
        return {**tx_data, 'signature': signature}

File: sdk/go/wallet/wallet.go
package wallet

import (
    "crypto/rand"
    "crypto/sha256"
    "encoding/hex"
    "fmt"
)

type Wallet struct {
    PrivateKey string
    Address    string
}

func Create() (*Wallet, error) {
    raw := make([]byte, 32)
    _, err := rand.Read(raw)
    if err != nil {
        return nil, err
    }
    h := sha256.Sum256(raw)
    return &Wallet{
        PrivateKey: hex.EncodeToString(raw),
        Address:    hex.EncodeToString(h[:])[:40],
    }, nil
}

func SignTransaction(privateKey string, tx map[string]any) (map[string]any, error) {
    // Simplified deterministic signature for SDK sample.
    payload := privateKey
    for k, v := range tx {
        payload += k + ":" + fmt.Sprintf("%v", v) + ";"
    }
    sig := sha256.Sum256([]byte(payload))
    out := map[string]any{}
    for k, v := range tx {
        out[k] = v
    }
    out["signature"] = hex.EncodeToString(sig[:])
    return out, nil
}
